src/oauth2-client/formats/siren.ts

```diff
@@ -1,9 +1,20 @@
 import { App } from '../../principal/types';
 
-export function newClient(user: App) {
+type NewClientQuery = {
+ clientId?: string;
+ allowGrantTypes?: string;
+ redirectUris?: string;
+ requirePkce?: string;
+}
+
+export function newClient(user: App, query: NewClientQuery ) {
 
   const memberHref = `/app/${user.id}`;
 
+  const redirectUris = query.redirectUris ? (query.redirectUris + '').split(',').join('\n') : '';
```

Comment:
Why the string concatenation here and for `allowGrantTypes`? Are these coming in as a different type and you're convert them to strings?

Reply:
Yeah, without them I get a `TypeError: query.redirectUris.split is not a function`
Any better ways you would suggest to eliminate the error perhaps?